delivery.yaml

```diff
@@ -0,0 +1,42 @@
+build_steps:
+    - desc: Prepare Environment
+      cmd: |
+        apt-get update
+        apt-get install -y python3 python3-pip jq
+        pip3 install scm-source
+
+    - desc: Install Docker
+      cmd: 'curl -sSL https://get.docker.com/ | sh'
+
+    - desc: scm-source file create
+      cmd: |
+        scm-source
+
+    - desc: Build and push docker images
+      cmd: |
+        # Please bump PATCH_VERSION if you change etcd.py or Dockerfile
+        PATCH_VERSION=p21
+        STOP_VERSION=2.3
+
+        ETCD_VERSIONS=$(curl -sL "https://api.github.com/repos/coreos/etcd/releases?per_page=100" | jq -r .[].name | sed -n 's/^v\([^-]*\)$/\1/p' | sort -urV)
+        ETCD_MAJOR_VERSIONS=$(sed 's/\.[0-9]*$//g' <<< "$ETCD_VERSIONS" | sort -urV)
+
+        for major_version in $ETCD_MAJOR_VERSIONS; do
+            version=$(egrep -m 1 "^${major_version/\./\\.}\." <<< "$ETCD_VERSIONS")
+
+            if [[ ! -z $prev ]]; then
+                IMAGE="registry-write.opensource.zalan.do/acid/etcd-cluster:$prev-$PATCH_VERSION"
+                docker build --build-arg ETCDVERSION_PREV=$version --build-arg ETCDVERSION=$prev -t $IMAGE .
+
+                # push docker images only for commits to the master branch
+                if [[ "x${CDP_SOURCE_BRANCH}" == "x" && "x${CDP_TARGET_BRANCH}" == "xmaster" ]]; then
+                    docker push $IMAGE
+                fi
```

Comment:
This will still build the image but discard it if the pipeline triggers from a PR.  Is this desired?

Reply:
Yep, I want to check that build succeeds before merging PR.